dora/src/semck/globaldef.rs

```diff
@@ -401,11 +424,7 @@ mod tests {
             pos(1, 15),
             SemError::ShadowStruct("Foo".into()),
         );
-        err(
-            "struct Foo {} fun Foo() {}",
-            pos(1, 15),
-            SemError::ShadowStruct("Foo".into()),
-        );
+        ok("struct Foo {} fun Foo() {}");
```

Comment:
Not sure this should be allowed: we would allow `class Foo {} fun Foo() {}`as well and then we don't know from `Foo()` whether we we should invoke the constructor/object or the function.

Reply:
Oh it would be the same for globals and vars as well, e.g. from `Foo(1)` we don't know whether we want to create an object or access an array element.

Reply:
You are right, I didn't intend to exclude struct; also right on the rest – I'll fix it. Thanks!

Reply:
@dinfuehr I pushed an updated PR.